### PR TITLE
feat: self contained styles asset-upload, nav, additional components with divs/spans

### PR DESCRIPTION
**Pull request by jbadan**
## Related Issue
SAP/fundamental-styles#136

closes https://github.com/SAP/fundamental-styles/issues/300

BREAKING CHANGE - removed `fd-modal—actions`, `fd-segmented-button`
## Description
Took another pass a scs changes with some lessons learned. Added resets to several components.
Fixed the spinner on the docs site.

Tests changed due to:
1) side-nav had hard coded styles including widths + margins in the playground - removed these causing huge diffs
2) calendar had blue days of the week - this is incorrect.
    - note: our calendar doesn't match still , but is getting closer 
3) popover placeholder changes - nothing to do with actual component

wiki page for calendar:
Screen Shot 2019-08-31 at 7 33 10 PM

## Screenshots
updated screenshots post pr feedback: 
Screen Shot 2019-09-03 at 2 50 20 PM
Screen Shot 2019-09-03 at 2 50 34 PM
Screen Shot 2019-09-03 at 2 50 40 PM
Screen Shot 2019-09-03 at 2 50 46 PM
Screen Shot 2019-09-03 at 2 50 53 PM
Screen Shot 2019-09-03 at 2 50 58 PM
Screen Shot 2019-09-03 at 2 51 03 PM

### CODE REVIEW

File: docs/pages/layouts/panel.md [hunk 1]
```diff
@@ -16,7 +16,8 @@ Place patterns and interactions within panels on your pages to achieve focus and
 > {{ site.data.strings.headerDisclaimer }}
 
 ## Elements
-
+- `.fd-panel__header`: Panel header.
+- `.fd-panel__head`: Container for title and description.
 - `.fd-panel__title`: Title of the panel.
 - `.fd-panel__description`: Title of the description.
 - `.fd-panel__actions`: Panel level actions such as add, remove, delete, sort, etc.
```
Comment (bcullman) [inline docs/pages/layouts/panel.md]: `fd-.panel__actions` still used?

File: scss/components/tile.scss [hunk 2]
```diff
@@ -79,6 +81,7 @@ $block: #{$fd-namespace}-tile;
     @include fd-reset-spacing();
```
Comment (bcullman) [inline scss/components/tile.scss]: is this still a needed mixin()?

File: scss/components/button-group.scss [hunk 3]
```diff
@@ -7,12 +7,9 @@
     .fd-button
 */
 $block: #{$fd-namespace}-button-group;
-$block-alt: #{$fd-namespace}-segmented-button;
```
[the remark this answers is not in the document]
Comment (jbadan) [reply scss/components/button-group.scss]: Yeah I didn't find it anywhere so I figured 🔥

File: scss/components/nav.scss [hunk 4]
```diff
@@ -22,12 +22,16 @@ $block: #{$fd-namespace}-nav;
...
+    @include fd-reset();
+  }
   &__link {
+    @include fd-reset();
+    @include fd-var-color("color", fd-color(action, 1), --fd-color-action-1);
     display: block;
     padding: $fd-nav-link-padding-y $fd-nav-link-padding-x;
     @include fd-selected() {
-      @include fd-var-color("color", $fd-color, --fd-color-text-1);
+      @include fd-var-color("color", fd-color(text, 1), --fd-color-text-1);
```
[the remark this answers is not in the document]
Comment (jbadan) [reply scss/components/nav.scss]: lol yeah I have no idea - I couldn't find it anywhere but it only took about 5 minutes to self contain so I figured why not. I'm fine with deleting it. @droshev any context on what this component is supposed to be?

File: scss/components/tile.scss (hunk 2 above)
Comment (jbadan) [reply scss/components/tile.scss]: Between spacing, type, color and it's other includes, it covered everything inside fd-reset already.

File: scss/components/panel.scss [hunk 5]
```diff
@@ -1,3 +0,0 @@
-/*
-see layouts/panel
-*/
```
[the remark this answers is not in the document]
Comment (jbadan) [reply scss/components/panel.scss]: Yeah so our clean-css postcss plugin deletes these comments anyways so we're just ending up with empty files. In the rebuild pipeline story, I plan on renaming all the "layout" components that we want to export to non-underscore names ie `layouts/_panel` to `layouts/panel`. I could do that there or here, doesn't matter.

[the remark this answers is not in the document]
Comment (greg-a-smith) [reply scss/components/panel.scss]: No problem then -- let's let some of the pipeline work take care of this.

File: scss/components/calendar.scss [hunk 6]
```diff
@@ -152,10 +154,12 @@ $fd-calendar-table-width: $fd-calendar-table-item-width * 7 !default;
     }
   }
   &__content {
+    @include fd-reset();
```
[the remark this answers is not in the document]
Comment (jbadan) [reply scss/components/calendar.scss]: added - in a really gross way. `__item` was actually being used to pass down all the styles to the `__text`, but the reset will reset all of that. This should be fundamentally reworked, but I think we should leave that to a followup to actually match the spec.

File: docs/pages/layouts/panel.md (hunk 1 above)
Comment (jbadan) [reply docs/pages/layouts/panel.md]: It is - we should have a followup to refine how flexbox is being used for the header, head, title, description and actions part of this. I think with some re-work a few of these could be eliminated - at least head and actions.

**Pull request merged**